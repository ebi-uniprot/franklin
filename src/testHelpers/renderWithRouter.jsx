import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

export default (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
};
