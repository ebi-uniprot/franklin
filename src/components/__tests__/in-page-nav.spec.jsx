import renderWithRouter from '../../testHelpers/renderWithRouter';
import InPageNav from '../in-page-nav';

describe('InPageNav component', () => {
  beforeAll(() => {
    global.IntersectionObserver = class IntersectionObserver {
      // eslint-disable-next-line class-methods-use-this
      observe() {
        return null;
      }

      // eslint-disable-next-line class-methods-use-this
      unobserve() {
        return null;
      }
    };
  });
  test('should render', () => {
    const sections = [
      {
        id: 'id1',
        label: 'First link',
      },
      {
        id: 'id2',
        label: 'Second link',
      },
      {
        id: 'id3',
        label: 'Third link',
        disabled: true,
      },
    ];

    const { asFragment } = renderWithRouter(<InPageNav sections={sections} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
