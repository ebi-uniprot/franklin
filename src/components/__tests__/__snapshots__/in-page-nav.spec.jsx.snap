// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InPageNav component should render 1`] = `
<DocumentFragment>
  <ul
    class="in-page-nav"
  >
    <li>
      <a
        href="/id1"
      >
        First link
      </a>
    </li>
    <li>
      <a
        href="/id2"
      >
        Second link
      </a>
    </li>
    <li
      class="disabled"
    >
      <a
        href="/id3"
      >
        Third link
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
